pkg/encoding/kzgEncoder/encoder.go

```diff
@@ -153,6 +164,25 @@ func ReadG2Point(n uint64, g *KzgConfig) (bls.G2Point, error) {
 	return g2point[0], nil
 }
 
+// Read g2 points from power of 2 file
+func ReadG2PointOnPowerOf2(exponent uint64, g *KzgConfig) (bls.G2Point, error) {
+
```

Comment:
Check `exponent` is less than log(SRSOrder)? It would be better than checking power is less than SRSOrder as it's before potential overflow